apps/frontend/src/pages/explore/explore.tsx

```diff
@@ -5,39 +5,95 @@ import { DataStatus } from "~/libs/enums/enums.js";
 import {
 	useAppDispatch,
 	useAppSelector,
+	useCallback,
 	useEffect,
 } from "~/libs/hooks/hooks.js";
-import { actions as exploreActions } from "~/modules/explore/explore.js";
+import { type Location } from "~/libs/types/types.js";
+import {
+	actions as exploreActions,
+	getRoutes,
+	loadMoreRoutes,
+} from "~/modules/explore/explore.js";
 import { actions as locationActions } from "~/modules/location/location.js";
 
 import { RoutesPanel } from "./libs/components/components.js";
 import { mockPOIs } from "./mock-pois.js";
 import styles from "./styles.module.css";
 
+const DEFAULT_PAGE_INCREMENT = 1;
+
 const Explore = (): React.JSX.Element => {
 	const dispatch = useAppDispatch();
 	const routesDataStatus = useAppSelector((state) => state.explore.dataStatus);
 	const routesError = useAppSelector((state) => state.explore.error);
 	const routes = useAppSelector((state) => state.explore.routes);
+	const hasMore = useAppSelector((state) => state.explore.hasMore);
+	const isLoadingMore = useAppSelector((state) => state.explore.isLoadingMore);
+	const loadMoreFailed = useAppSelector(
+		(state) => state.explore.loadMoreFailed,
+	);
+	const page = useAppSelector((state) => state.explore.page);
+	const searchTerm = useAppSelector((state) => state.explore.searchTerm);
```

Comment:
we have access to search value inside routes panel, so we can just pass it there in onLoadMore. this way we won't need to store it store and reset/control it
```suggestion
onLoadMore({ search })
```